frontend/web/components/release-pipelines/SinglePipelineStageAction.tsx

```diff
@@ -24,42 +25,65 @@ const getSegment = (action: StageActionRequest | undefined) => {
 }
 
 interface SinglePipelineStageActionProps {
+  indexPos?: number
   action?: StageActionRequest
-  onActionChange: (option: { value: string; label: string }) => void
-  onSegmentChange: (option: { value: number; label: string }) => void
+  headerText?: string
+  onRemoveAction?: (id: number) => void
+  onActionChange: (option: { value: string; label: string }, id: number) => void
+  onSegmentChange: (
+    option: { value: number; label: string },
+    id: number,
+  ) => void
   isSegmentsLoading: boolean
   segmentOptions: { label: string; value: number }[] | undefined
 }
 
 const SinglePipelineStageAction = ({
   action,
+  headerText = 'Then',
+  indexPos = 0,
   isSegmentsLoading,
   onActionChange,
+  onRemoveAction,
   onSegmentChange,
   segmentOptions,
 }: SinglePipelineStageActionProps) => {
   return (
     <>
       <Row>
         <div className='flex-1 and-divider__line' />
-        <div className='mx-2'>Then</div>
+        <div className='mx-2'>{headerText}</div>
         <div className='flex-1 and-divider__line' />
       </Row>
-      <FormGroup>
+      <FormGroup className='mt-2'>
         <InputGroup
           title='Flag Action'
           component={
-            <Select
-              menuPortalTarget={document.body}
-              maxMenuHeight={120}
-              value={Utils.toSelectedValue(
-                getActionType(action),
-                FLAG_ACTION_OPTIONS,
-                { label: 'Select an action', value: '' },
+            <div className='d-flex align-items-center'>
+              <div className='w-100'>
+                <Select
+                  menuPortalTarget={document.body}
+                  maxMenuHeight={120}
+                  value={Utils.toSelectedValue(
+                    getActionType(action),
+                    FLAG_ACTION_OPTIONS,
+                    { label: 'Select an action', value: '' },
+                  )}
+                  options={FLAG_ACTION_OPTIONS}
```

Comment:
<img width="373" height="333" alt="image" src="https://github.com/user-attachments/assets/89996ca0-dbab-4d04-a090-ef325c1dbff7" />

Detail too but could be nice to disable the `Enable/disable flag for everyone` once it has already been selected in an action